### PR TITLE
Adding tests for core utilities Parameter class

### LINKED ISSUE #61: Add unit tests for Parameter in core utilites
Parameter class needs unit tests

### PR DESCRIPTION
**Pull request by cainja**
Closes #61.

### CODE REVIEW

File: test/arcade/core/util/ParameterTest.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        }
+    }
+
+    @Test
+    public void constructor_isFracAndMuGreaterThan1_throwsIllegalArgumentException() {
+        MersenneTwisterFast random = new MersenneTwisterFast();
+        double mu = 2.0;
+        double sigma = 0.25;
+
+        IllegalArgumentException e =
```
Comment (jessicasyu) [inline test/arcade/core/util/ParameterTest.java:127]: Can simplify to:

```java
IllegalArgumentException e =
        assertThrows(
                IllegalArgumentException.class,
                () -> new Parameter(mu, sigma, true, random));
```

File: test/arcade/core/util/ParameterTest.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                assertThrows(
+                        IllegalArgumentException.class,
+                        () -> {
+                            new Parameter(mu, sigma, true, random);
+                        });
+        assertEquals("Fractional parameter must be between 0 and 1", e.getMessage());
+    }
+
+    @Test
+    public void constructor_isFracAndMuLessThan0_throwsIllegalArgumentException() {
```
Comment (jessicasyu) [inline test/arcade/core/util/ParameterTest.java:137]: Can we put all the constructor tests at the top?

File: test/arcade/core/util/ParameterTest.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        double mu = -2.0;
+        double sigma = 0.25;
+
+        IllegalArgumentException e =
+                assertThrows(
+                        IllegalArgumentException.class,
+                        () -> {
+                            new Parameter(mu, sigma, true, random);
+                        });
+        assertEquals("Fractional parameter must be between 0 and 1", e.getMessage());
```
Comment (jessicasyu) [inline test/arcade/core/util/ParameterTest.java:148]: I don't love testing for the exact exception message. Thoughts on something like a custom `OutOfBoundsException` that extends `IllegalArgumentException`?

File: test/arcade/core/util/ParameterTest.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ParameterTest {
+    @Test
+    public void testConstructor_populatesAttributes() {
+        MersenneTwisterFast random = new MersenneTwisterFast();
+        Parameter p = new Parameter(4.0, 0.25, false, random);
+
+        assertAll(
+                "Parameter",
```
Comment (jessicasyu) [inline test/arcade/core/util/ParameterTest.java:14]: Thoughts on defaulting to having the `assertAll` description match the test name?

File: test/arcade/core/util/ParameterTest.java [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+                () -> assertEquals(4.0, p.getMu(), 0.00001),
+                () -> assertEquals(1.0, p.getSigma(), 0.00001));
+    }
+
+    @Test
+    public void update_returnsParameter() {
```
Comment (jessicasyu) [inline test/arcade/core/util/ParameterTest.java:20]:
```suggestion
    public void update_called_updatesParameterAttributes() {
```

**Review by jessicasyu — APPROVED** [review]

File: test/arcade/core/util/ParameterTest.java (hunk 4 above)
Comment (cainja) [reply test/arcade/core/util/ParameterTest.java:14]: It feels ike redundant info? What about just removing for obvious cases?

File: test/arcade/core/util/ParameterTest.java (hunk 3 above)
Comment (cainja) [reply test/arcade/core/util/ParameterTest.java:148]: I feel like I'm adding an unnecessary class to maintain for a single use exception, but I'll add if a project goal is more robust error handling during runtime.

**Comment by cainja** [comment]
Latest commits reflect proposed changes for active conversations.

**Pull request merged**